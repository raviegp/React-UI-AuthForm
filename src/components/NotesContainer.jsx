import React from 'react';
import Grid from '@material-ui/core/Grid';
import { withStyles } from '@material-ui/core/styles';
import Note from './Note';
import PropTypes from 'prop-types';

const styles = theme => ({
    root: {
        marginTop: theme.spacing.unit * 2,
    },
    spaceAround: {
        display: 'flex',
        justifyContent: 'space-around',
        marginBottom: theme.spacing.unit * 2,
    },
});

const NotesContainer = (props) => {
    const { 
            classes, 
            notes, 
            handleRemoveNote,
            searchTerm,
        }  = props;
    return (
        <div className={classes.root}>
            <Grid container spacing={8}>
                {    
                    notes.map(note => (
                        <Grid item xs={12} sm={4} key={note.id} className={classes.spaceAround}>
                            <Note 
                                note={note}
                                handleRemoveNote={handleRemoveNote}
                                searchTerm={searchTerm}
                            />
                        </Grid>
                    ))
                }
            </Grid>
        </div>
    );
}

NotesContainer.propTypes = {
    classes: PropTypes.object.isRequired,
    notes: PropTypes.array.isRequired,
    handleRemoveNote: PropTypes.func.isRequired,
    searchTerm: PropTypes.string,
};

export default withStyles(styles)(NotesContainer);