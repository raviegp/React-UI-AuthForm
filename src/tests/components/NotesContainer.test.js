import NotesContainer from '../../components/NotesContainer';
import React from 'react';
import { shallow, mount } from 'enzyme';
import notes from '../fixtures/notes';
import { createSerializer } from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

let wrapper, handleRemoveNote;
beforeEach(() => {
    handleRemoveNote = jest.fn();
    wrapper = shallow(
        <NotesContainer 
            notes={notes}
            handleRemoveNote={handleRemoveNote}
        />
    );
});

test('Render NotesContainer component with notes correctly', () => {
    expect(wrapper).toMatchSnapshot();
});