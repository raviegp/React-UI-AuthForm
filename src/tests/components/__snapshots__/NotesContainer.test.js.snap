// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NotesContainer component correctly 1`] = `
<NotesContainer
  classes={
    Object {
      "root": "NotesContainer-root-1",
      "spaceAround": "NotesContainer-spaceAround-2",
    }
  }
  handleRemoveNote={[MockFunction]}
  notes={
    Array [
      Object {
        "id": 1,
        "noteDescription": "Note 1 description",
        "noteTitle": "Note 1",
      },
      Object {
        "id": 2,
        "noteDescription": "Note 2 description",
        "noteTitle": "Note 2",
      },
      Object {
        "id": 3,
        "noteDescription": "Note 3 description",
        "noteTitle": "Note 3",
      },
      Object {
        "id": 4,
        "noteDescription": "Note 4 description",
        "noteTitle": "Note 4",
      },
    ]
  }
/>
`;

exports[`Render NotesContainer component with notes correctly 1`] = `
<NotesContainer
  classes={
    Object {
      "root": "NotesContainer-root-1",
      "spaceAround": "NotesContainer-spaceAround-2",
    }
  }
  handleRemoveNote={[MockFunction]}
  notes={
    Array [
      Object {
        "id": 1,
        "noteDescription": "Note 1 description",
        "noteTitle": "Note 1",
      },
      Object {
        "id": 2,
        "noteDescription": "Note 2 description",
        "noteTitle": "Note 2",
      },
      Object {
        "id": 3,
        "noteDescription": "Note 3 description",
        "noteTitle": "Note 3",
      },
      Object {
        "id": 4,
        "noteDescription": "Note 4 description",
        "noteTitle": "Note 4",
      },
    ]
  }
/>
`;
